apps/src/music/views/InstrumentGrid/index.tsx

```diff
@@ -44,7 +51,9 @@ const InstrumentGrid: React.FunctionComponent<Props> = ({
   const [currentValue, setCurrentValue] = useState(initialValue);
   const [isLoading, setIsLoading] = useState(false);
   const [currentPreviewTick, setCurrentPreviewTick] = useState(0);
-  const [scaleMode, setScaleMode] = useState<ScaleMode>('simple');
+
+  const scaleMode = currentValue.scaleMode;
+  const key = MusicRegistry.player.getKey();
 
   useEffect(() => {
     onChange(currentValue);
```

Comment:
Overall thought - it might be simpler to handle the relative to absolute conversion as we move between react and blockly instead of having to convert and un-convert and various points in the view. I think that should look like 1) convert from relative (Blockly field) to absolute (React state) when setting the initial state as the `currentValue` and 2) convert from absolute (React state) to relative (Blockly field) when writing the updated value here, in the `onChange` callback.

Reply:
This component does now convert an incoming value to absolute, and an outgoing value to relative.  As discussed separately, it might be possible to convert this component to operate entirely in relative pitch values, but likely out of scope for this change.